new/property.md

```diff
@@ -0,0 +1,206 @@
+---
+RFC: unassigned
+Author: John Keiser <jkeiser@chef.io>
+Status: Draft
+Type: Standards Track
+---
+
+# Resource Properties
+
+We add `property` DSL to resources, similar to (and interoperable with) LWRP `attribute`.
+
+It works very similarly to attribute syntax.  There are no backwards compatibility issues with this proposal, as all `attribute` functionality remains the same.
+
+## Motivation
+
+    As a Chef user,
+    I want resource attributes and node attributes to use different words,
+    so that they don't lead me to conflate them as concepts.
+
+## Specification
+
+### Defining properties
+
+Resource class definitions may now call `property` to create a resource property.  This works similarly to LWRP attribute, but with some important additions and differences.
+
+```ruby
+class MyResource < Chef::Resource
+  property :path, Path, name_attribute: true
+  property :content, String
+end
+```
+
+This calls MyResource.properties[name] = <property type>.
+
+When a property is defined this way:
+- `my_resource.<name>` will get the property value.
+- `my_resource.<name> = <value>` will set the property value.
+- `my_resource.<name> <value>` will set the property value.
+- `MyResource.state_attrs` will include the property name.
+- `MyResource.properties` will contain the property type.
+- The setter and getter manipulate the class variable `@<name>`.
+
+#### Use `property` instead of `attribute` in documentation
+
+`attribute` will continue to be supported; there are simply too many things in the world to deprecate it. However, any generic documentation that talks about attributes will be renamed to talk about properties. `attribute` itself will still be documented.
+
+### Setting properties
+
+### nil
+
+In order to allow for `nil` as a potential explicit value, property setters accept `nil` and set the value of the property to `nil`.  This differs from `attribute`, which considers `property(nil)` to be a get.
```

Comment:
Oh, now I understand what you mean by "core chef attributes"--you mean attributes in core Chef resources?  Yes, we'll have to have an RFC to be able to convert existing attributes over, since it will cause `file.mode(nil)` to emit an error (you can't set `mode` to `nil`, it's not a valid value).  We *can* convert them over if we implement this difference as a modifier like `nil_is_default: true` and add the modifier to the core Chef codebase; but I'm not sure that's worth it when we could just RFC it.